src/main/java/com/sap/cloudfoundry/client/facade/adapters/LogCacheClient.java

```diff
@@ -18,120 +22,67 @@
-
 public class LogCacheClient {
 
-    private static final ObjectMapper MAPPER = new ObjectMapper().configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false)
-                                                                 .configure(DeserializationFeature.UNWRAP_ROOT_VALUE, true);
-
-    private static final Logger LOGGER = LoggerFactory.getLogger(LogCacheClient.class);
-
-    private final String logCacheApi;
     private final OAuthClient oAuthClient;
     private final Map<String, String> requestTags;
+    private final ConnectionContext connectionContext;
+
+    private static final String SOURCE_TYPE_KEY_NAME = "source_type";
+    private static final int MAX_LOG_COUNT = 1000;
 
-    public LogCacheClient(String logCacheApi, OAuthClient oAuthClient, Map<String, String> requestTags) {
-        this.logCacheApi = logCacheApi;
+    public LogCacheClient(OAuthClient oAuthClient, Map<String, String> requestTags, ConnectionContext connectionContext) {
         this.oAuthClient = oAuthClient;
         this.requestTags = requestTags;
+        this.connectionContext = connectionContext;
     }
 
     public List<ApplicationLog> getRecentLogs(UUID applicationGuid, LocalDateTime offset) {
-        HttpResponse<InputStream> response = CloudUtil.executeWithRetry(() -> executeRequest(applicationGuid, offset));
-
-        return parseBody(response.body()).getLogs()
-                                         .stream()
-                                         .map(this::mapToAppLog)
-                                         // we use a linked list so that the log messages can be a LIFO sequence
-                                         // that way, we avoid unnecessary sorting and copying to and from another collection/array
-                                         .collect(LinkedList::new, LinkedList::addFirst, LinkedList::addAll);
-    }
-
-    private HttpResponse<InputStream> executeRequest(UUID applicationGuid, LocalDateTime offset) {
-        try {
-            HttpRequest request = buildGetLogsRequest(applicationGuid, offset);
-            LOGGER.info(Messages.CALLING_LOG_CACHE_ENDPOINT_TO_GET_APP_LOGS);
-            var response = CloudFoundryClientFactory.HTTP_CLIENT.send(request, BodyHandlers.ofInputStream());
-            if (response.statusCode() / 100 != 2) {
-                var status = HttpStatus.valueOf(response.statusCode());
-                throw new CloudOperationException(status, status.getReasonPhrase(), parseBodyToString(response.body()));
-            }
-            LOGGER.info(Messages.APP_LOGS_WERE_FETCHED_SUCCESSFULLY);
-            return response;
-        } catch (IOException | InterruptedException e) {
-            throw new CloudException(e.getMessage(), e);
-        }
+        org.cloudfoundry.logcache.v1.LogCacheClient logCacheClient = createReactorLogCacheClient();
+        ReadResponse applicationLogsResponse = readApplicationLogs(logCacheClient, applicationGuid, offset);
```

Comment:
now we're creating a new client for every get recent logs call? Is this operation expensive? Are there worse/better results in stress/load tests?

Reply:
This change doesn't have negative impact on the performance